scanner/src/main/kotlin/scanners/ScanCode.kt

```diff
@@ -128,16 +135,37 @@ class ScanCode(config: ScannerConfiguration) : LocalScanner(config) {
     }
 
     override fun bootstrap(): File {
-        val gitRoot = File(".").searchUpwardsForSubdirectory(".git")
-        val scancodeDir = File(gitRoot, "scanner/src/funTest/assets/scanners/scancode-toolkit")
-        if (!scancodeDir.isDirectory) throw IOException("Directory '$scancodeDir' not found.")
+        // Use the .zip file despite it being slightly larger than the .tar.bz2 file as the latter for some reason does
+        // not complete to unpack on Windows.
+        val url = "https://github.com/nexB/scancode-toolkit/releases/download/v$scannerVersion/" +
+                "scancode-toolkit-$scannerVersion.zip"
+
+        log.info { "Downloading $this from '$url'... " }
+
+        val request = Request.Builder().get().url(url).build()
+
+        return OkHttpClientHelper.execute(HTTP_CACHE_PATH, request).use { response ->
+            val body = response.body()
+
+            if (response.code() != HttpURLConnection.HTTP_OK || body == null) {
+                throw IOException("Failed to download $this from $url.")
+            }
+
+            if (response.cacheResponse() != null) {
+                log.info { "Retrieved $this from local cache." }
+            }
 
-        val configureExe = if (OS.isWindows) "configure.bat" else "configure"
-        val configurePath = File(scancodeDir, configureExe)
-        ProcessCapture(configurePath.absolutePath, "--clean").requireSuccess()
-        ProcessCapture(configurePath.absolutePath).requireSuccess()
+            val scannerArchive = createTempFile(suffix = url.substringAfterLast("/"))
+            Okio.buffer(Okio.sink(scannerArchive)).use { it.writeAll(body.source()) }
 
-        return scancodeDir
+            val unpackDir = createTempDir()
```

Comment:
Prior to this commit there was just one directory holding the scan code executable.
With this change: each instance of ScanCode will have its own (temp) directory holding the scanner zip contents.
For 'production' code this seems to be no change as there is only max one instance created per process.
I wonder what effect this has on test execution (speed), have you considered that?

Reply:
That's a valid concern that's common to all bootstrapped scanners. We should bootstrap to a directory (e.g. within the current user's home) from where we can re-use a previously bootstrapped scanner. But that's a change I wanted to do later for all scanners at once.

Reply:
Agree, makes all sense.